internal/pkg/service/stream/definition/sink_base.go

```diff
@@ -11,10 +12,14 @@ type Sink struct {
 	Versioned
 	Switchable
 	SoftDeletable
-	Type        SinkType   `json:"type" validate:"required,oneof=table"`
-	Name        string     `json:"name" validate:"required,min=1,max=40"`
-	Description string     `json:"description,omitempty" validate:"max=4096"`
-	Table       *TableSink `json:"table" validate:"required_if=Type table"`
+	Type        SinkType             `json:"type" validate:"required,oneof=table"`
+	Name        string               `json:"name" validate:"required,min=1,max=40"`
+	Description string               `json:"description,omitempty" validate:"max=4096"`
+	Config      configpatch.PatchKVs `json:"config"` // see stream/config/config.Patch
```

Comment:
Sink now contains configuration patch as KV pairs. 

So we can modify anything provided by the `config.Patch`, but it is not directry coupled with the `config` package.